roachtest: sqlalchemy failed

roachtest.sqlalchemy [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/11564547?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/11564547?buildTab=artifacts#/sqlalchemy) on release-23.1 @ [953350ffadaebf2e33f4aff3cf95a5e2e8b69d5f](https://github.com/cockroachdb/cockroach/commits/953350ffadaebf2e33f4aff3cf95a5e2e8b69d5f):


```
(orm_helpers.go:212).summarizeFailed: 
Tests run on Cockroach v23.1.9-dev-953350ffadaebf2e33f4aff3cf95a5e2e8b69d5f
Tests run against sqlalchemy 2.0.2
494 Total Tests Run
493 tests passed
1 test failed
645 tests skipped
0 tests ignored
0 tests passed unexpectedly
1 test failed unexpectedly
0 tests expected failed but skipped
0 tests expected failed but not run
---
--- FAIL: test/test_suite_sqlalchemy.py::ComponentReflectionTest_cockroachdb+psycopg2_9_5_0::test_get_multi_columns (unexpected)
For a full summary look at the sqlalchemy artifacts
An updated blocklist (sqlAlchemyBlocklist) is available in the artifacts' sqlalchemy log
test artifacts and logs in: /artifacts/sqlalchemy/run_1
```
<p>Parameters: <code>ROACHTEST_arch=amd64</code>
, <code>ROACHTEST_cloud=gce</code>
, <code>ROACHTEST_cpu=4</code>
, <code>ROACHTEST_encrypted=false</code>
, <code>ROACHTEST_ssd=0</code>
</p>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/p/roachfana/teamcity-11564547-1693460547-134-n1cpu4/1693504224775/1693504346093)

</p>
</details>
/cc @cockroachdb/sql-foundations
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*sqlalchemy.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-31113

roachtest.sqlalchemy [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/11581541?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/11581541?buildTab=artifacts#/sqlalchemy) on release-23.1 @ [d762d323b40b09feba4604e5c9c475614fd21b0c](https://github.com/cockroachdb/cockroach/commits/d762d323b40b09feba4604e5c9c475614fd21b0c):


```
(orm_helpers.go:212).summarizeFailed: 
Tests run on Cockroach v23.1.9-dev-d762d323b40b09feba4604e5c9c475614fd21b0c
Tests run against sqlalchemy 2.0.2
494 Total Tests Run
493 tests passed
1 test failed
645 tests skipped
0 tests ignored
0 tests passed unexpectedly
1 test failed unexpectedly
0 tests expected failed but skipped
0 tests expected failed but not run
---
--- FAIL: test/test_suite_sqlalchemy.py::ComponentReflectionTest_cockroachdb+psycopg2_9_5_0::test_get_multi_columns (unexpected)
For a full summary look at the sqlalchemy artifacts
An updated blocklist (sqlAlchemyBlocklist) is available in the artifacts' sqlalchemy log
test artifacts and logs in: /artifacts/sqlalchemy/run_1
```
<p>Parameters: <code>ROACHTEST_arch=amd64</code>
, <code>ROACHTEST_cloud=gce</code>
, <code>ROACHTEST_cpu=4</code>
, <code>ROACHTEST_encrypted=false</code>
, <code>ROACHTEST_ssd=0</code>
</p>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/p/roachfana/teamcity-11581541-1693547132-125-n1cpu4/1693589748778/1693589864098)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #109909 roachtest: sqlalchemy failed [C-test-failure O-roachtest O-robot T-sql-foundations branch-release-23.1.9-rc release-blocker]
- #109888 roachtest: sqlalchemy failed [C-test-failure O-roachtest O-robot T-sql-foundations branch-release-22.2 release-blocker]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*sqlalchemy.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/cmd/roachtest/tests/knex.go
+++ b/pkg/cmd/roachtest/tests/knex.go
@@ -63,7 +63,7 @@ func registerKnex(r registry.Registry) {
 			c,
 			node,
 			"add nodesource repository",
-			`sudo apt install ca-certificates && curl -fsSL https://deb.nodesource.com/setup_16.x | sudo -E bash -`,
+			`sudo apt install ca-certificates && curl -fsSL https://deb.nodesource.com/setup_18.x | sudo -E bash -`,
 		)
 		require.NoError(t, err)
 

--- a/pkg/cmd/roachtest/tests/nodejs_postgres.go
+++ b/pkg/cmd/roachtest/tests/nodejs_postgres.go
@@ -73,7 +73,7 @@ func registerNodeJSPostgres(r registry.Registry) {
 			c,
 			node,
 			"add nodesource repository",
-			`sudo apt install ca-certificates && curl -sL https://deb.nodesource.com/setup_16.x | sudo -E bash -`,
+			`sudo apt install ca-certificates && curl -sL https://deb.nodesource.com/setup_18.x | sudo -E bash -`,
 		)
 		require.NoError(t, err)
 

--- a/pkg/cmd/roachtest/tests/sequelize.go
+++ b/pkg/cmd/roachtest/tests/sequelize.go
@@ -96,7 +96,7 @@ func registerSequelize(r registry.Registry) {
 			c,
 			node,
 			"add nodesource repository",
-			`sudo apt install ca-certificates && curl -sL https://deb.nodesource.com/setup_16.x | sudo -E bash -`,
+			`sudo apt install ca-certificates && curl -sL https://deb.nodesource.com/setup_18.x | sudo -E bash -`,
 		); err != nil {
 			t.Fatal(err)
 		}

--- a/pkg/cmd/roachtest/tests/sqlalchemy.go
+++ b/pkg/cmd/roachtest/tests/sqlalchemy.go
@@ -30,7 +30,7 @@ import (
 var sqlAlchemyResultRegex = regexp.MustCompile(`^(?P<test>test.*::.*::[^ \[\]]*(?:\[.*])?) (?P<result>\w+)\s+\[.+]$`)
 var sqlAlchemyReleaseTagRegex = regexp.MustCompile(`^rel_(?P<major>\d+)_(?P<minor>\d+)_(?P<point>\d+)$`)
 
-var supportedSQLAlchemyTag = "2.0.2"
+var supportedSQLAlchemyTag = "2.0.20"
 
 // This test runs the SQLAlchemy dialect test suite against a single Cockroach
 // node.

--- a/pkg/cmd/roachtest/tests/typeorm.go
+++ b/pkg/cmd/roachtest/tests/typeorm.go
@@ -90,7 +90,7 @@ func registerTypeORM(r registry.Registry) {
 			c,
 			node,
 			"add nodesource repository",
-			`sudo apt install ca-certificates && curl -fsSL https://deb.nodesource.com/setup_16.x | sudo -E bash -`,
+			`sudo apt install ca-certificates && curl -fsSL https://deb.nodesource.com/setup_18.x | sudo -E bash -`,
 		); err != nil {
 			t.Fatal(err)
 		}

